lib/fuzzyDateTime.js

```diff
@@ -0,0 +1,99 @@
+import { getLanguage } from './common.js';
+import { formatDateTime, formatDate } from './dateTime.js';
+import { millisecondsPer } from './constants.js';
+
+const { second, minute, hour, day } = millisecondsPer;
+const thirtySeconds = 30 * second;
+const halfHour = 30 * minute;
+const hourAndHalf = 90 * minute;
+const fortyFiveMinutes = 45 * minute;
+const threeAndAHalfDays = 84 * hour;
+const sixHours = 6 * hour;
+
+const defaultRtf = Intl && Intl.RelativeTimeFormat &&
+	new Intl.RelativeTimeFormat(getLanguage(), {
+		localeMatcher: 'best fit',
+		numeric: 'auto',
+		style: 'long'
+	});
+
+const midnightTime = dateTime =>
+	new Date(
+		dateTime.getFullYear(),
+		dateTime.getMonth(),
+		dateTime.getDate()
+	).getTime();
+
+const noonTime = dateTime =>
+	new Date(
+		dateTime.getFullYear(),
+		dateTime.getMonth(),
+		dateTime.getDate(),
+		12
+	).getTime();
+
+// Adapted from:
+// https://search.d2l.dev/xref/lms/lp/framework/web/D2L.LP.Web/UI/JavaScript/Globalization/Formatting/DateTime/DateTimeFormatter.js?r=438b5dc3#125
+//
+// but modified to work with:
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RelativeTimeFormat
+//
+const formatFuzzy = formatFunction => (inputDate, {
+	absoluteFormat = 'short',
+	origin = new Date(),
+	onUpdate = null,
+	rtf = defaultRtf
+} = {}) => {
+	const midnight = midnightTime(origin);
+	const noon = noonTime(origin);
+	const now = origin.getTime();
+
+	const sameDay = midnightTime(inputDate) === midnight;
+	const timespan = inputDate.getTime() - origin.getTime();
+	const timespanFromNoon = inputDate.getTime() - noon;
+	const timespanAbs = Math.abs(timespan);
+	const isPast = timespan < 0;
+
+	let text;
+	let timeout;
+	if (rtf && timespanAbs < thirtySeconds) {
+		text = rtf.format(Math.round(timespan / second), 'second');
+		timeout = timespanAbs * (isPast ? 1 : 0.5);
```

Comment:
Yes, I was trying to avoid the lang term and take advantage of rtf. Currently it backs off by doubling, so you see `1, 2, 4, 8, 16` and past 30 seconds you get `1 minute ago`

Changing to `(isPast ? 3 : 0.5)` you get `1, 4, 16`

Reply:
What do you think about just setting the timeout to however long it would take to get to the "1 minute ago" threshold? So if it was "5 seconds ago" the timeout would be 25 seconds.

Reply:
Sounds good. What about the other direction? It reads `in 1 minute` and the time is 36 seconds in the future? Wait 66 seconds to update to `1 minute ago`, bypassing the fine grained seconds rendering